integration-cli/docker_cli_build_test.go

```diff
@@ -5115,3 +5115,32 @@ func TestBuildNotVerbose(t *testing.T) {
 
 	logDone("build - not verbose")
 }
+
+func TestBuildVolumesFiltered(t *testing.T) {
+	defer func() {
+		deleteAllContainers()
+		deleteImages("test")
+	}()
+
+	ctx, err := fakeContext(`
+	FROM busybox
+	VOLUME /foo
+	RUN [ -d /foo ]
+	RUN touch /foo/bar
+	RUN [ -f /foo/bar ]
+	ADD baz /foo/baz
+	RUN [ -f /foo/baz ]
+	`,
+		map[string]string{"baz": "qux"})
+	defer ctx.Close()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err = buildImageFromContext("test", ctx, false); err != nil {
+		t.Fatalf("expected build to succeed:\n%v", err)
+	}
```

Comment:
Ah never mind, I forgot that each RUN will be like a new container from the previous image